src/linker/default_environment.yaml

```diff
@@ -0,0 +1,17 @@
+# Default values used as necessary
+computing_environment: local
+container_engine: undefined
+slurm:
+  account: proj_simscience
```

Comment:
Does this change _require_ that we have defaults for everything? I don't think we want a default here...

Reply:
I flip-flopped on this for a bit. It seems like this might be a user-facing (ish, via docs) file where they're free to update as desired so I stuck it in there.

Happy to remove this if you think it's better to simply not have any defaults.

Reply:
In fact, if you don't want it to be so user-facing, I can remove the new `default_environment.yaml` and just implement it as a dict on the configuration module.

Reply:
I like having defaults for most things where we can at least make a guess, but I think the slurm account and and queue shouldn't have defaults.

Reply:
I don't have any preference about whether it is a file or a dictionary in code.

Reply:
Done